src/python/pants/engine/internals/engine_test.py

```diff
@@ -703,175 +704,171 @@ def a_rule(input: ComplicatedInput) -> Output:
     return Output(snapshot_1=input.snapshot_1, snapshot_2=input.snapshot_2)
 
 
-class MoreComplicatedEngineAware(TestBase):
-    @classmethod
-    def rules(cls):
-        return (
-            *super().rules(),
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=[
             a_rule,
             QueryRule(Output, (ComplicatedInput,)),
-        )
-
-    def test_more_complicated_engine_aware(self) -> None:
-        tracker = WorkunitTracker()
-        handler = StreamingWorkunitHandler(
-            self.scheduler,
-            callbacks=[tracker.add],
-            report_interval_seconds=0.01,
-            max_workunit_verbosity=LogLevel.TRACE,
-        )
-        with handler.session():
-            input_1 = CreateDigest(
-                (
-                    FileContent(path="a.txt", content=b"alpha"),
-                    FileContent(path="b.txt", content=b"beta"),
-                )
+            QueryRule(ProcessResult, (Process,)),
+        ],
+        isolated_local_store=True,
+    )
+
+
+def test_more_complicated_engine_aware(rule_runner: RuleRunner) -> None:
```

Comment:
The diff algorithm is not producing useful line-by-line diffs below, just FYI. The actual change is just converting the methods to standalone functions and calling methods on `rule_runner` instead of `self`.